db/src/db.rs

```diff
@@ -689,130 +762,67 @@ impl DB {
-                    };
-
-                    let a: i64 = match a_ {
-                        &entmod::Entid::Entid(ref a__) => *a__,
-                        &entmod::Entid::Ident(ref a__) => *self.schema.require_entid(&a__.to_string())?,
-                    };
-
-                    let attribute: &Attribute = self.schema.require_attribute_for_entid(&a)?;
-                    if attribute.fulltext {
-                        bail!(ErrorKind::NotYetImplemented(format!("Transacting :db/fulltext entities is not yet implemented: {:?}", entity)))
-                    }
-
-                    // This is our chance to do schema-aware typechecking: to either assert that the
-                    // given value is in the attribute's value set, or (in limited cases) to coerce
-                    // the value into the attribute's value set.
-                    let typed_value: TypedValue = self.to_typed_value(v_, &attribute)?;
-
-                    let added = true;
-                    if attribute.multival {
-                        non_fts_many.push((e, a, typed_value, added));
-                    } else {
-                        non_fts_one.push((e, a, typed_value, added));
-                    }
-                    Ok(())
-                },
-
-                Entity::AddOrRetract {
-                    op: OpType::Retract,
-                    e: entmod::EntidOrLookupRef::Entid(ref e_),
-                    a: ref a_,
-                    v: entmod::ValueOrLookupRef::Value(ref v_) } => {
-
-                    let e: i64 = match e_ {
-                        &entmod::Entid::Entid(ref e__) => *e__,
-                        &entmod::Entid::Ident(ref e__) => *self.schema.require_entid(&e__.to_string())?,
-                    };
-
-                    let a: i64 = match a_ {
-                        &entmod::Entid::Entid(ref a__) => *a__,
-                        &entmod::Entid::Ident(ref a__) => *self.schema.require_entid(&a__.to_string())?,
-                    };
-
-                    let attribute: &Attribute = self.schema.require_attribute_for_entid(&a)?;
-                    if attribute.fulltext {
-                        bail!(ErrorKind::NotYetImplemented(format!("Transacting :db/fulltext entities is not yet implemented: {:?}", entity)))
-                    }
-
-                    // This is our chance to do schema-aware typechecking: to either assert that the
-                    // given value is in the attribute's value set, or (in limited cases) to coerce
-                    // the value into the attribute's value set.
-                    let typed_value: TypedValue = self.to_typed_value(v_, &attribute)?;
-
-                    let added = false;
+    /// Update the current partition map materialized view.
+    // TODO: only update changed partitions.
+    pub fn update_partition_map(&self, conn: &rusqlite::Connection) -> Result<()> {
```

Comment:
Is it sensible to use `&mut rusqlite::Connection` here as a signal that this is a mutating operation?